cognitive/src/main/python/synapse/ml/services/langchain/LangchainTransform.py

```diff
@@ -214,11 +243,36 @@ def udfFunction(x):
                 openai.api_key = self.getSubscriptionKey()
                 openai.api_base = self.getUrl()
                 openai.api_version = self.getApiVersion()
-            return self.getChain().run(x)
+
+            error_messages = {
+                openai.error.Timeout: "OpenAI API request timed out, please retry your request after a brief wait and contact us if the issue persists: {}",
+                openai.error.APIError: "OpenAI API returned an API Error: {}",
+                openai.error.APIConnectionError: "OpenAI API request failed to connect, check your network settings, proxy configuration, SSL certificates, or firewall rules: {}",
+                openai.error.InvalidRequestError: "OpenAI API request was invalid: {}",
+                openai.error.AuthenticationError: "OpenAI API request was not authorized, please check your API key or token and make sure it is correct and active. You may need to generate a new one from your account dashboard: {}",
+                openai.error.PermissionError: "OpenAI API request was not permitted, make sure your API key has the appropriate permissions for the action or model accessed: {}",
+                openai.error.RateLimitError: "OpenAI API request exceeded rate limit: {}",
+            }
+
+            try:
+                result = self.getChain().run(x)
+                error_message = ""
+            except tuple(error_messages.keys()) as e:
+                result = ""
+                error_message = error_messages[type(e)].format(e)
+
+            return result, error_message
 
         outCol = self.getOutputCol()
+        errorCol = self.getErrorCol()
         inCol = dataset[self.getInputCol()]
-        return dataset.withColumn(outCol, udfFunction(inCol))
+
+        return (
+            dataset.withColumn("result", udfFunction(inCol))
```

Comment:
sorry i just noticed this, name the column result_{uid} so that this doesent break if someone already has a column called result